src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/glide/MauiCustomTarget.java

```diff
@@ -12,20 +12,22 @@
 import com.bumptech.glide.request.transition.Transition;
 import com.microsoft.maui.ImageLoaderCallback;
 
-public class MauiCustomTarget extends CustomTarget<Drawable>
-{
+public class MauiCustomTarget extends CustomTarget<Drawable> {
     private final ImageLoaderCallback callback;
     private final RequestManager requestManager;
+    private boolean completed;
 
-    public MauiCustomTarget(ImageLoaderCallback callback, RequestManager requestManager)
-    {
+    public MauiCustomTarget(ImageLoaderCallback callback, RequestManager requestManager) {
+        this.completed = false;
         this.callback = callback;
         this.requestManager = requestManager;
     }
 
     @Override
     public void onLoadFailed(@Nullable Drawable errorDrawable) {
-        super.onLoadFailed(errorDrawable);
+        if (completed)
+            return;
+        this.completed = true;
```

Comment:
This is the real fix - for both the `onLoadFailed` and `onResourceReady` methods. For both `MauiCustomTarget` and `MauiCustomViewTarget`. 

In no world is calling this on the same instance valid because the ctor takes a callback which is wrapping a `TaskCompletionSource`. Even if we use the `TryXxx` methods, this means nothing because there is work that is done after the await - and if nothing is awaiting, then nothing will get done.